pdq/abstractness.go

```diff
@@ -0,0 +1,23 @@
+package pdq
+
+import (
+	"errors"
+)
+
+func CalcAbstractness(abstractElementsNum int, concreteElementsNum int) (float64, error) {
+	if abstractElementsNum < 0 {
+		return -1, errors.New("the number of abstract elements must not be a negative number")
+	}
+
+	if concreteElementsNum < 0 {
+		return -1, errors.New("the number of concrete elements must not be a negative number")
+	}
+
+	if abstractElementsNum == 0 || concreteElementsNum == 0 {
+		return 0, nil
+	}
```

Comment:
Initially I entered the wrong formula, but in the current one if number of abstract elements is 0, then the formula works, but always returns 0(which is correct). 

If only number of concrete elements is 0, then the division by 0 should happen (abstractElementsNum/concreteElementsNum). This could represent a theoretical situation where a software is composed entirely of abstract elements with no concrete implementations. 

I don't want to throw out an error right away, so i handle this case and return abstractElementsNum, signaling that the abstraction is very high.